x-pack/plugins/ml/server/routes/modules.ts

```diff
@@ -85,9 +85,29 @@ export function dataRecognizer({ router, mlLicense }: RouteInitialization) {
    *
    * @api {get} /api/ml/modules/recognize/:indexPatternTitle Recognize index pattern
    * @apiName RecognizeIndex
-   * @apiDescription Returns the list of modules that matching the index pattern.
-   *
-   * @apiParam {String} indexPatternTitle Index pattern title.
+   * @apiDescription By supplying an index pattern, discover if any of the modules are a match for data in that index.
+   * @apiParam {String} indexPatternTitle index pattern to recognize. Note that this does not need to be a Kibana
+   *      index pattern, and can be the name of a single Elasticsearch index,
+   *      or include a wildcard (*) to match multiple indices.
```

Comment:
please use `@apiSchema (params) indexPatternTitleSchema` instead

Reply:
It requires the following:
1. Defined the schema with a unique name in `x-pack/plugins/ml/server/routes/schemas/modules.ts` like this 
```typescript
export const recognizeIndexParamsSchema = schema.object({
  /**
   * index pattern to recognize. Note that this does not need to be a Kibana
   * index pattern, and can be the name of a single Elasticsearch index,
   * or include a wildcard (*) to match multiple indices.
   */
  indexPatternTitle: schema.string(),
})
```
2. import this schema into this file (`x-pack/plugins/ml/server/routes/modules.ts`) and replace the current inline definition with `recognizeIndexParamsSchema`.
3. update annotation to `@apiSchema (params) recognizeIndexParamsSchema`

Reply:
Done in https://github.com/elastic/kibana/pull/66738/commits/7a6fd145038f2be82eebe24681815f7ca03d38cc